sherpa/astro/ui/tests/test_astro_session.py

```diff
@@ -0,0 +1,26 @@
+#
+#  Copyright (C) 2016  Smithsonian Astrophysical Observatory
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+from sherpa.astro.ui.utils import Session
+
+
+# bug #303
+def test_show_bkg_model():
```

Comment:
Is it worth adding another test where the id is set explicitly in the call? e.g.

```
def test_show_bkg_model_id():
    session = Session()
    session.show_bkg_model()
    session.show_bkg_model("xx")
```

I ask because I see slightly-different behavior when no data files have been loaded (this is with the current master branch):

```
% python
Python 3.5.2 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:53:06) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sherpa.astro import ui
WARNING: imaging routines will not be available, 
failed to import sherpa.image.ds9_backend due to 
'RuntimeErr: DS9Win unusable: Could not find xpaget on your PATH'
>>> ui.show_bkg_model()
>>> ui.show_bkg_model('xx')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<string>", line 1, in show_bkg_model
  File "/lagado.real/sherpa/sherpa-294/sherpa/astro/ui/utils.py", line 542, in show_bkg_model
    all += self._get_show_bkg_model(id, bkg_id)
  File "/lagado.real/sherpa/sherpa-294/sherpa/astro/ui/utils.py", line 364, in _get_show_bkg_model
    bkg_ids.extend(self._background_sources.get(id, {}).keys())
AttributeError: 'dict_keys' object has no attribute 'extend'
```

Ideally we'd also have a test that included a case where a background model was set up. From a quick look it doesn't look like we have many existing tests - e.g. onlu `sherpa/astro/ui/tests/test_chips_unit.py` and `sherpa/astro/datastack/tests/test_datastack.py` seem to use the `set*_bkg_model`, and `sherpa/astro/ui/tests/test_serialize.py` use `set_bkg_source`